File: BackEnd/eMenka/eMenka.Domain/Classes/Brand.cs
using System.Collections.Generic;

namespace eMenka.Domain.Classes
{
    public class Brand
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public ICollection<Vehicle> Vehicles { get; set; }
        public ICollection<Model> Models { get; set; }
        public ICollection<Series> Series { get; set; }
        public ICollection<EngineType> EngineTypes { get; set; }
        public ICollection<ExteriorColor> ExteriorColors { get; set; } = new List<ExteriorColor>();
        public ICollection<InteriorColor> InteriorColors { get; set; } = new List<InteriorColor>();
    }
}

File: eFenkaFrontend/src/app/models/brand/brand.ts
import { Model } from '../model/model';
import { Serie } from '../serie/serie';
import { ExteriorColor } from '../exterior-color/exterior-color';
import { InteriorColor } from '../interior-color/interior-color';
import { EngineType } from '../engine-type/engine-type';

export class Brand {
  id: number;
  name: string;
  models: Model[];
  series: Serie[];
  engineTypes: EngineType[];
  exteriorColors: ExteriorColor[];
  interiorColors: InteriorColor[];

  constructor(id: number, name: string) {
    this.id = id;
    this.name = name;
  }
}
